assets/actions/package.go

```diff
@@ -0,0 +1,258 @@
+	if err != nil {
+		log.Printf("Error connecting to Meshery Cloud: %v\n", err)
+		return nil
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Printf("Error reading response body: %v\n", err)
+		return nil
+	}
+	return body
+}
+
+func processPattern(pattern CatalogPattern, ghAccessToken string) error {
+	patternImageURL := getPatternImageURL(pattern)
+	patternType := getPatternType(string(pattern.CatalogData.Type))
+	patternInfo := getStringOrEmpty(pattern.CatalogData.PatternInfo)
+	patternCaveats := getStringOrEmpty(pattern.CatalogData.PatternCaveats)
+
+	compatibilityStrings := make([]string, len(pattern.CatalogData.Compatibility))
+	for i, v := range pattern.CatalogData.Compatibility {
+		compatibilityStrings[i] = string(v)
+	}
+	compatibility := getCompatibility(compatibilityStrings)
+
+	dir := filepath.Join("..", "..", "collections", "_catalog", patternType)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		os.MkdirAll(dir, 0755)
+	}
+
+	if err := writePatternFile(pattern, patternType, patternInfo, patternCaveats, compatibility, patternImageURL); err != nil {
+		return err
+	}
+	invokeGitHubAction(pattern.ID, patternImageURL, ghAccessToken)
+
+	return nil
+}
+
+func getPatternImageURL(pattern CatalogPattern) string {
+	var imageURL string
+	if pattern.CatalogData.SnapshotURL == nil {
+		imageURL = fmt.Sprintf("https://raw.githubusercontent.com/layer5labs/meshery-extensions-packages/master/action-assets/design-assets/%s-light.png,https://raw.githubusercontent.com/layer5labs/meshery-extensions-packages/master/action-assets/design-assets/%s-dark.png", pattern.ID, pattern.ID)
+	} else {
+		if len(pattern.CatalogData.SnapshotURL) == 1 {
+			imageURL = pattern.CatalogData.SnapshotURL[0]
+		} else if len(pattern.CatalogData.SnapshotURL) > 1 {
+			imageURL = strings.Join(pattern.CatalogData.SnapshotURL, ",")
+		}
+	}
+	return imageURL
+}
+
+func getPatternType(patternType string) string {
+	if patternType == "" {
+		patternType = "Deployment"
+	}
+	return strings.ToLower(strings.ReplaceAll(patternType, " ", "-"))
+}
+
+func getStringOrEmpty(value string) string {
+	if value == "" {
+		return "\"\""
+	}
+	return value
+}
+
+func getCompatibility(compatibility []string) string {
+	var compatLines []string
+	for _, compat := range compatibility {
+		compatLines = append(compatLines, fmt.Sprintf("    - %s", compat))
+	}
+	return strings.Join(compatLines, "\n")
+}
+
+func writePatternFile(pattern CatalogPattern, patternType, patternInfo, patternCaveats, compatibility, patternImageURL string) error {
+	dir := filepath.Join("..", "..", mesheryCatalogFilesDir, pattern.ID)
+	deployFilePath := filepath.Join(dir, "deploy.yml")
+	os.MkdirAll(dir, 0755)
+	ioutil.WriteFile(deployFilePath, []byte(pattern.PatternFile), 0644)
+
+	deployContent, err := ioutil.ReadFile(deployFilePath)
+	if err != nil {
+		return fmt.Errorf("Failed to read file: %v", err)
```

Comment:
Always reuse or define a new Meshkit error,
I believe an error is already defined for reading a file in the `meshkit` package.